tests/gem5/checkpoint-tests/test-checkpoints.py

```diff
@@ -0,0 +1,266 @@
+)
+
+gem5_verify_config(
+    name="test-checkpoint-x86-hello-save-checkpoint",
+    fixtures=(),
+    verifiers=(save_checkpoint_verifier,),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "x86-hello-save-checkpoint.py",
+    ),
+    config_args=[
+        "--checkpoint-path",
+        joinpath(resource_path, "x86-hello-test-checkpoint"),
+    ],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+gem5_verify_config(
+    name="test-checkpoint-x86-hello-restore-checkpoint",
+    fixtures=(),
+    verifiers=(hello_verifier,),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "x86-hello-restore-checkpoint.py",
+    ),
+    config_args=[],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+gem5_verify_config(
+    name="test-checkpoint-x86-fs-save-checkpoint",
+    fixtures=(),
+    verifiers=(save_checkpoint_verifier,),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "x86-fs-save-checkpoint.py",
+    ),
+    config_args=[
+        "--checkpoint-path",
+        joinpath(resource_path, "x86-fs-test-checkpoint"),
+    ],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+gem5_verify_config(
+    name="test-checkpoint-x86-fs-restore-checkpoint",
+    fixtures=(),
+    verifiers=(),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "x86-fs-restore-checkpoint.py",
+    ),
+    config_args=[],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+gem5_verify_config(
+    name="test-checkpoint-power-hello-save-checkpoint",
+    fixtures=(),
+    verifiers=(save_checkpoint_verifier,),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "power-hello-save-checkpoint.py",
+    ),
+    config_args=[
+        "--checkpoint-path",
+        joinpath(resource_path, "power-hello-test-checkpoint"),
+    ],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+gem5_verify_config(
+    name="test-checkpoint-power-hello-restore-checkpoint",
+    fixtures=(),
+    verifiers=(hello_verifier,),
+    config=joinpath(
+        config.base_dir,
+        "tests",
+        "gem5",
+        "checkpoint-tests",
+        "power-hello-restore-checkpoint.py",
+    ),
+    config_args=[],
+    valid_isas=(constants.all_compiled_tag,),
+    valid_hosts=constants.supported_hosts,
+    length=constants.quick_tag,
+)
+
+# gem5_verify_config(
```

Comment:
Add a link here to the issue you created/will create to not this bug (also link the issue back to this PR).

Reply:
Should I remove the mips and sparc test files?

Reply:
Why would you remove those?

Reply:
I thought of removing mips because it does not support checkpoints. I don't know if the support will be added.

Reply:
If mips genuinely doesn't support checkpoints then you're fine to remove these tests. I want tests there for stuff that is supposed to be supported but is broken.

Reply:
+1 to remove MIPS

Reply:
Just removed the mips test, I will add the link to the issue when I make one.